src/main/java/hunternif/mc/atlas/core/BiomeDetectorBase.java

```diff
@@ -63,7 +64,13 @@ int priorityForBiome(Biome biome) {
 	public int getBiomeID(Chunk chunk) {
 		int biomeCount = Biome.REGISTRY.getKeys().size();
 
-		int[] chunkBiomes = ByteUtil.unsignedByteToIntArray(chunk.getBiomeArray());
+		int[] chunkBiomes;

```

Comment:
I've seen the following lines at least thrice, please make a function for that. 
`util/BiomUtil::getBiomeArray()` sounds like a good place

Reply:
I actually would like advice on this. It would be best to have something of the form:
private static final Function<?> getBiomeArrayFunc;
This would allow us to cache the result of getMethod() rather than calling it every time. I couldn't get it to work though.

Reply:
final might now work in this context actually, which may have been my issue.

Reply:
As far as I'm concerned, the changes made in a22a5a4 sort this. BiomeDectorNether and BiomeDetectorEnd are already just copy/pasted from BiomeDetectorBase, and desperately need generalising. However, I feel that should be the subject of its own PR.

Reply:
Looks okay now. I agree with the BiomeDetector refactoring.